File: LeetCode/1822. Sign of the Product of an Array/solution.py
class Solution:
    def arraySign(self, nums: List[int]) -> int:
        def func(x):
            if x > 0:
                return 1
            elif x == 0:
                return 0
            return -1
        cur = 1
        for x in nums:
            cur *= func(x)
        return cur 

File: LeetCode/1822. Sign of the Product of an Array/solution.cpp
class Solution {
public:
    int arraySign(vector<int>& nums) {
        int cur = 1;
        for (auto &x: nums)
        {
            int tmp = x >= 0 ? (x > 0 ? 1: 0) : -1;
            cur *= tmp;
        }
        return cur;
    }
};